### PR TITLE
PLUI-518 - Users should see accurate VM networking info

### PR DESCRIPTION
**Pull request by brnewby602**
* add options parameter to listPorts and listFloatingIPs to allow filters to be applied on response data
* add test to ensure passing in options doesnt error the request
* add getArgsWithCallback util function to separate callback and any nbr of optional params
* add tests for util function to ensure it provides valid responses for less, equal, and more than expected arguments
* update README

### CODE REVIEW

File: lib/neutron.js [hunk 1]
```diff
@@ -273,12 +273,25 @@ Neutron.prototype.createFloatingIp = function(floating_network_id, cb){
...
+//    'device_id': 'abcdef123567'
+//  }
+//}
+Neutron.prototype.listFloatingIps = function(options, cb){
   var self = this;
   var ip_array = [];
   var n = 0;
   var request_options = this.getRequestOptions('/floatingips', true);
+  // get the actual args passed in as this function can be overloaded
+  var args = osutils.getArgsWithCallback(this.listFloatingIps.length, Array.prototype.slice.call(arguments));
```
Comment (BannedInSweden) [inline lib/neutron.js]: optionally move Array.prototype.slice.call

File: lib/os-utils.js [hunk 2]
```diff
@@ -245,10 +245,39 @@ function getRequestError(error, response, body)
...
+//I.E.:
+//  getArgsWithCallback(2, ["param1"])
+//  will return [undefined, "param1"]
+ function getArgsWithCallback(expected_length, arg_array) {
+  var return_args = [];
+  var i = 0;
+
+  if(Array.isArray(arg_array))
+  {
+    var cb = arg_array.pop();
```
Comment (BannedInSweden) [inline lib/os-utils.js]: declare cb at top (can still pop and set here)

File: test/os-utils-test.js [hunk 3]
```diff
@@ -39,3 +39,32 @@ exports.getError = {
...
+    var param2 = function(){};
+    var args = utils.getArgsWithCallback(2, [param1, param2]);
+    test.deepEqual(args, [param1, param2], 'Should return same arguments as passed in');
+    test.done();
+  },
+  confirmResponseWithLessArgs: function(test)
+  {
+    var param1 = function(){};
+    var args = utils.getArgsWithCallback(2, [param1]);
+    test.deepEqual(args, [undefined, param1], 'Should return array of same length with undefined elements for missing arguments');
```
Comment (BannedInSweden) [inline test/os-utils-test.js]: should params not passed set as null not undefined?

File: lib/neutron.js (hunk 1 above)
Comment (brnewby602) [inline lib/neutron.js]: done

File: lib/os-utils.js (hunk 2 above)
Comment (brnewby602) [inline lib/os-utils.js]: done

File: test/os-utils-test.js (hunk 3 above)
Comment (brnewby602) [inline test/os-utils-test.js]: done

**Pull request merged**